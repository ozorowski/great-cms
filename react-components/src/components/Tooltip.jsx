import React from 'react';
import './stylesheets/Tooltip.scss'

export default class Tooltip extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const DynamicHeadingTag = `h${this.props.priority}`;
        return (
            <div id={this.props.id} role="tooltip" className={`tooltip ${this.props.className}`}>
                <DynamicHeadingTag className="tooltip__heading h-m">{this.props.heading}</DynamicHeadingTag>
                {this.props.body.map((item, key) =>
                    <p className="tooltip__paragraph" key={key}>{item}</p>
                )}
            </div>
        )
    }

}