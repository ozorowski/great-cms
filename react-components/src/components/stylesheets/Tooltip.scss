@import 'helpers/colours';

.tooltip {
    border-radius: 0.3em;
    background: $white;
    box-shadow: 0 0 3px rgba(0,0,0,0.3);
    padding: 20px;
    max-width: 50%;
    min-width: 150px;
    position: absolute;
    z-index: 2;

    background-image: url('/static/img/placeholder.png');
    background-position: right bottom;
    background-repeat: no-repeat;
    background-size: 30%;
}

.tooltip__heading {
    padding-top: 0;
}

.tooltip__paragraph {
    padding-right: 30%;

    &:last-child {
        margin-bottom: 0;
    }
}