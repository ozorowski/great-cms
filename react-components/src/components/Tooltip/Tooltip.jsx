import React from 'react';
import './Tooltip.scss'

export default class Tooltip extends React.Component {

    constructor(props) {
        super(props);
        this.tooltipRef = React.createRef();
    }

    componentDidMount() {
        //dom operation moved here rather than render function for performance
        this.tooltipRef.current.appendChild(this.props.tooltipContent);
    }

    render() {
        return (
            <div id={this.props.id}
                role="tooltip"
                className={`tooltip ${this.props.className}`}
                ref={this.tooltipRef}>
            </div>
        )
    }
}