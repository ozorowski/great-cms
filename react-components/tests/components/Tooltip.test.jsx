import React from 'react'
import { mount } from 'enzyme'
import Tooltip from '@src/components/Tooltip/Tooltip';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Tooltip component', () => {
    Enzyme.configure({ adapter: new Adapter() })

    test('Matches the snapshot', () => {
        const props = {
            tooltipContent: document.createElement('div')
        };

        const wrapper = mount(<Tooltip {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
