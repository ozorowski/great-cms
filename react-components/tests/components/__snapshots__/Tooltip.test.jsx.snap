// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component Matches the snapshot 1`] = `ReactWrapper {}`;
